src/serlo-editor-repo/plugin-text/utils/list.ts

```diff
@@ -1,32 +1,58 @@
 import { ListsEditor, ListType } from '@prezly/slate-lists'
-import { Editor as SlateEditor } from 'slate'
+import { Element, Editor as SlateEditor } from 'slate'
+import { ReactEditor } from 'slate-react'
 
-import { selectionHasElement } from './selection'
+import { existsInAncestors } from './document'
 
-export function isOrderedListActive(editor: SlateEditor) {
-  return selectionHasElement((e) => e.type === 'ordered-list', editor)
+export function isElementWithinList(element: Element, editor: SlateEditor) {
+  return existsInAncestors(
+    (elem) => elem.type === 'unordered-list' || elem.type === 'ordered-list',
+    { location: ReactEditor.findPath(editor, element) },
+    editor
+  )
 }
 
-export function isUnorderedListActive(editor: SlateEditor) {
-  return selectionHasElement((e) => e.type === 'unordered-list', editor)
+export function isSelectionWithinList(
+  editor: SlateEditor,
+  listType?: 'unordered-list' | 'ordered-list'
+) {
+  if (!editor.selection) return false
+
+  return existsInAncestors(
+    (element) => {
+      return listType
+        ? element.type === listType
+        : element.type === 'ordered-list' || element.type === 'unordered-list'
```

Comment:
Maybe consider using the enums from [this file](https://github.com/serlo/frontend/blob/staging/src/serlo-editor-repo/plugin-text/types/text-editor.ts#LL31C16-L31C16) instead of string literals.

(I know string literals were used before, but that's old code 😄 )

Reply:
I agree, that will be less vulnerable to sneaky typos. :)